Source/WebCore/platform/graphics/ImageObserver.h

```diff
@@ -54,6 +57,7 @@ class ImageObserver : public RefCounted<ImageObserver>, public CanMakeWeakPtr<Im
     virtual void imageFrameAvailable(const Image&, ImageAnimatingState, const IntRect* changeRect = nullptr, DecodingStatus = DecodingStatus::Invalid) = 0;
     virtual void changedInRect(const Image&, const IntRect* changeRect = nullptr) = 0;
     virtual void scheduleRenderingUpdate(const Image&) = 0;
+    virtual SingleThreadWeakHashSet<CachedImageClient> visibleClientsInViewport(const Image&, const Document&) { return { }; }
```

Comment:
Rather than return a `SingleThreadWeakHashSet<CachedImageClient>` from these various new functions, you could just call it `likelyVisibleInViewportMoreThanOnce` or something, and do the counting in `CachedImage::visibleClientsInViewport()`.

Reply:
A different API named `clientsIntersectInViewport()` was added. This API returns `bool`.

Reply:
Another approach was implemented. See the commit message.